assets/js/theme.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+(() => {
+    const LS_THEME_KEY = 'theme';
+    const THEMES = Object.freeze({
+        LIGHT: 'light',
+        DARK: 'dark',
+        AUTO: 'auto'
+    });
+
+    const body = document.body;
+    const config = body.getAttribute("data-theme");
+
+    const getThemeState = () => {
+        const lsState = localStorage.getItem(LS_THEME_KEY);
+        if (lsState) return lsState;
+
+        let state;
+        switch (config) {
+            case THEMES.DARK:
+                state = THEMES.DARK;
+                break;
+            case THEMES.LIGHT:
+                state = THEMES.LIGHT;
+                break;
+            case THEMES.AUTO:
+            default:
+                state = window.matchMedia("(prefers-color-scheme: dark)").matches
+                    ? THEMES.DARK
+                    : THEMES.LIGHT;
+                break;
+        }
+        return state;
+    };
+
+    const initTheme = (state) => {
+        if (state === THEMES.DARK) {
+            document.documentElement.classList.add(THEMES.DARK);
+            document.documentElement.classList.remove(THEMES.LIGHT);
+        } else if (state === THEMES.LIGHT) {
+            document.documentElement.classList.remove(THEMES.DARK);
+            document.documentElement.classList.add(THEMES.LIGHT);
+        }
+    };
+
+    // init theme ASAP, then do the rest.
+    initTheme(getThemeState());
+    requestAnimationFrame(() => body.classList.remove("notransition"))
+
+    const toggleTheme = () => {
+        const state = getThemeState();
+        if (state === THEMES.DARK) {
+            localStorage.setItem(LS_THEME_KEY, THEMES.LIGHT);
+            initTheme(THEMES.LIGHT);
+        } else if (state === THEMES.LIGHT) {
+            localStorage.setItem(LS_THEME_KEY, THEMES.DARK);
+            initTheme(THEMES.DARK);
+        }
+    }
+
+    window.addEventListener('DOMContentLoaded', () => {
+        // Theme switch
+        const lamp = document.getElementById("mode");
+
+        lamp.addEventListener("click", () => toggleTheme());
+
+        // Blur the content when the menu is open
+        const cbox = document.getElementById("menu-trigger");
+
+        cbox.addEventListener("change", function () {
+            const area = document.querySelector(".wrapper");
+            this.checked
+                ? area.classList.add("blurry")
+                : area.classList.remove("blurry");
+        });
```

Comment:
This can be replaced with CSS.
Unfortunately `:has()` is still experimental and disabled by default in `firefox` and unavailable in `firefox android`.
See:  [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/:has#browser_compatibility) and [caniuse](https://caniuse.com/?search=%3Ahas)

So I'm not sure is this a good idea to use it.



```scss
// wrap with @media where #menu-trigger becomes visible
body:has(#menu-trigger:checked) {
    .wrapper {
      animation: 0.2s ease-in forwards blur;
      -webkit-animation: 0.2s ease-in forwards blur;
    }
}
```

Reply:
@kusyka911, first thanks for this PR.

Here is what I have to say regarding the use of CSS `:has()`:

![image](https://user-images.githubusercontent.com/122173059/236593749-847e143f-554f-41a5-99ca-5aa757807f23.png)

Thanks for the caniuse link. Looks like the global support for this property is only 86%.
As a safety measure, I have set the following bars for myself:
- global support > 95% - good to go
- global support >= 90% - we can consider using this
- global support <90 - we should not consider using this

In conclusion, I feel we should not use this. It's not worth sacrificing browser support for the few lines of JS.

What do you say @kusyka911 ?

Reply:
In general I agree, but please also do not forget about no-script environments :wink:
We can add polyfill but in this case it will also not work in no-script environments.
So I don't see any good reason to push this idea.

Reply:
> We can add polyfill but in this case it will also not work in no-script environments.

Nice observation 😄